src/main/java/gregtech/common/covers/detector/CoverDetectorBase.java

```diff
@@ -0,0 +1,91 @@
+package gregtech.common.covers.detector;
+
+import codechicken.lib.raytracer.CuboidRayTraceResult;
+import gregtech.api.cover.CoverBehavior;
+import gregtech.api.cover.ICoverable;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.EnumActionResult;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.text.TextComponentTranslation;
+
+public abstract class CoverDetectorBase extends CoverBehavior {
```

Comment:
Right now all the subclasses of this base class implement the `ITickable` interface. Would it make sense for this base class to implement the interface instead and move the common functionality here? (I think the common functionality would just be checking the offset timer though)

Reply:
I am aware of this but I don't feel that providing custom function which would do exactly same thing as `update` (except for offset timer check) is great idea. As the added values is too small to obscure fact that cover is ticking which can be easily recognized by the `update` function.